# Tarjetas

dataset: https://docs.google.com/spreadsheets/d/1E2QvH18DyNDw1eGqU40D_KHSEfSzbipV/edit?usp=sharing&ouid=115931210603772521641&rtpof=true&sd=true


In [2]:
import pandas as pd
import numpy
#!pip install openpyxl

In [3]:
%pwd

'C:\\Users\\alejo\\UCEMA\\poo_teorica\\poo_ejercicios\\04-transformaciones'

In [27]:
tarjetas = pd.read_csv('data/tarjetas.xlsx - Sheet1.csv', delimiter=',', index_col='Unnamed: 0')

In [28]:
tarjetas.head()

PreseTaciOn_Fecha        FeCha_de_Pago  cod_clearing  \
0  2022-06-15 00:00:00  2022-07-01 00:00:00          1009   
1  2022-06-15 00:00:00  2022-07-01 00:00:00          1011   
2  2022-06-15 00:00:00  2022-07-01 00:00:00          2031   
3  2022-06-15 00:00:00  2022-07-01 00:00:00          2040   
4  2022-06-15 00:00:00  2022-07-01 00:00:00          2800   

                                   desc_creaing  NRo_Liquidacion  \
0  VENTAS C/DESCUENTO CONTADO                              27822   
1  VENTAS C/DTO CUOTAS FINANC. OTORG.                      27822   
2  ARANCEL                                                 27822   
3                      PROMO CUOTAS AHORA 12/18            27822   
4  TOTAL DEDUCCIONES                                       27822   

   cod_sucursal  Cod_comercio  Cod_pagadora Cod_marca_producto  \
0           716      25107406           156       VISA CREDITO   
1           716      25107406           156       VISA CREDITO   
2           716      25107406           156       VISA CREDITO   
3           716      25107406           156       VISA CREDITO   
4           716      25107406           156       VISA CREDITO   

   importe_concepto  
0           5096.00  
1         413500.60  
2           7534.70  
3          18416.85  
4          49585.43

### 1- Pasar todos los nombres de las columnas a minusculas

In [7]:
#cambio el nombre de las columnas y lo guardo en la columna original
tarjetas.columns= tarjetas.columns.str.lower()

In [8]:
tarjetas.columns

Index(['presetacion_fecha', 'fecha_de_pago', 'cod_clearing', 'desc_creaing',
       'nro_liquidacion', 'cod_sucursal', 'cod_comercio', 'cod_pagadora',
       'cod_marca_producto', 'importe_concepto'],
      dtype='object')

### 2-Cuantas sucursales únicas hay? ¿Cuáles?

In [9]:
# metodo unique para obtener los elementos de la columna sucursales sin repeticion
tarjetas.cod_sucursal.unique()

array([716,  98, 671, 519, 241], dtype=int64)

### 3- Cuantas tarjetas de crédito distitas hay? ¿Cuáles?
¿Se te ocurre un nombre mejor para esa columna?

In [10]:
# metodo rename para renombrar, uso el parametro columns para renombrar la columna que se pasa por diccionario
tarjetas2 = tarjetas.rename(columns={'cod_marca_producto':'procesador_de_pago'})

In [30]:
# uso el metodo unique de nuevo para ver todas las procesadoras de pago
procesador = tarjetas2.procesador_de_pago.unique()
[print(i) for i in procesador]

VISA CREDITO
VISA DEBITO
MASTERCARD
MASTERCARD DEBIT
MAESTRO
ARGENCARD


[None, None, None, None, None, None]

In [13]:
# uso el metodo contains para traer de la columna procesador_de_pago todos los elementos que contengan 'CREDITO'
mask_credit_card = tarjetas2['procesador_de_pago'].str.contains(pat= ' CREDITO')
tarjetas2.procesador_de_pago[mask_credit_card].unique()


array(['VISA CREDITO'], dtype=object)

### 4- Cuants filas y columnas hay?


In [36]:
tarjetas2.shape

(47280, 10)

### 5- ¿Cuanto suma el total de importe_concepto?


In [37]:
total_importe= tarjetas2.importe_concepto.sum()

print(f'Suma total de importe_concepto: {total_importe}')

Suma total de importe_concepto: 4145618332.503


### 6- ¿Cuanto dinero se cobro por mes? Cree un lineplot para gráficarlo


In [38]:
#convierto la columna a datetime, infer_datetime_format te reconoce el formato en que viene la fecha
tarjetas2['fecha_de_pago'] =  pd.to_datetime(tarjetas2['fecha_de_pago'], infer_datetime_format=True)

In [48]:
#agrupo por mes y lo guardo en la variable by_month
by_month = tarjetas2.groupby(tarjetas2.fecha_de_pago.dt.month)

#con la agrupacion hago la suma de la columna importe_concepto y lo muestro en un frame
by_month.sum()['importe_concepto'].to_frame(name='suma mes')


suma mes
fecha_de_pago              
7              4.145618e+09

In [59]:
#Suma por dia
by_day = tarjetas2.groupby(tarjetas2.fecha_de_pago.dt.day)

suma_importe_dia= by_day.sum()['importe_concepto'].to_frame(name='suma dia')
suma_importe_dia

suma dia
fecha_de_pago              
1              1.473177e+08
4              1.976693e+08
5              7.024681e+08
6              1.002753e+08
7              1.116222e+08
8              1.465996e+08
11             2.870652e+08
12             2.403649e+08
13             1.227566e+08
14             1.293021e+08
15             1.469553e+08
18             2.730872e+08
19             2.219563e+08
20             1.363231e+08
21             1.431265e+08
22             1.531738e+08
25             2.727540e+08
26             2.242873e+08
27             1.235888e+08
28             1.294521e+08
29             1.354729e+08

### 7- Cuanto dinero se cobro por sucursal?


In [18]:
# chequeo cuantas sucursales hay
tarjetas2.cod_sucursal.unique()


array([716,  98, 671, 519, 241], dtype=int64)

In [46]:
# agrupo por sucursal y hago la suma de la columna importe_concepto
importe_sucursal = tarjetas2.groupby('cod_sucursal').sum()['importe_concepto'].to_frame(name='importe total')
importe_sucursal

importe total
cod_sucursal               
98             1.401451e+08
241            9.773019e+05
519            3.621125e+09
671            2.418829e+08
716            1.414878e+08

### 8- Cuanto dinero se cobro por mes por sucursal?


In [60]:
#creo variables con las columnas de fecha por mes y la de sucursal para usarlas en el groupby
month= tarjetas2.fecha_de_pago.dt.month
sucursal= tarjetas2.cod_sucursal

tarjetas2.groupby(by=[month, sucursal]).sum()['importe_concepto']

fecha_de_pago  cod_sucursal
7              98              1.401451e+08
               241             9.773019e+05
               519             3.621125e+09
               671             2.418829e+08
               716             1.414878e+08
Name: importe_concepto, dtype: float64

In [61]:
day= tarjetas2.fecha_de_pago.dt.day

tarjetas2.groupby(by=[day, sucursal]).sum()['importe_concepto']


fecha_de_pago  cod_sucursal
1              98              3.669431e+06
               519             1.312675e+08
               671             7.608022e+06
               716             4.772726e+06
4              98              6.478613e+06
                                   ...     
29             98              4.980121e+06
               241             2.472456e+04
               519             1.182347e+08
               671             6.924635e+06
               716             5.308723e+06
Name: importe_concepto, Length: 91, dtype: float64

### 9- Cuanto dinero se cobro por mes por sucursal y por descreaing?

In [62]:
tarjetas2.columns


Index(['presetacion_fecha', 'fecha_de_pago', 'cod_clearing', 'desc_creaing',
       'nro_liquidacion', 'cod_sucursal', 'cod_comercio', 'cod_pagadora',
       'procesador_de_pago', 'importe_concepto'],
      dtype='object')

In [63]:
descreaing= tarjetas2.desc_creaing

importe_month_sucursal_descreaing = tarjetas2.groupby(by=[month, sucursal, descreaing]).sum()['importe_concepto']


fecha_de_pago  cod_sucursal  desc_creaing                                
7              98             PERCEPCION IVA R.G. 2408   3,00 %              1.133729e+04
                             AJUSTE SIRTAC JUL. 1RA QUINC                    1.861809e+04
                             ARANCEL                                         4.826957e+05
                             CRED FISCAL AHORA 12/18 VTA EN CUOTAS 10,50%    9.107601e+04
                             CRED. FISCAL PAG.S/DTO.FIN.OTORG 21,00%         3.315445e+04
                                                                                 ...     
               716           SUBTOTAL NETO DE PAGOS                          3.279938e+07
                             TOTAL DEDUCCIONES                               3.392830e+06
                             TOTAL LIQUIDACION                               3.277539e+07
                             VENTAS C/DESCUENTO CONTADO                      1.665725e+07
                          

### 10- Pase a columnas los valores presentes en desc_creaing y ponga los importes como valores y en el indice la fecha de pago y el codigo de comercio

### 11- Quedarse solo con tarjeta visa. Crear una columna que sume 'VENTAS C/DESCUENTO CONTADO' con
                                'VENTAS C/DTO CUOTAS FINANC. OTORG.'

### 12- Quedarse solo con la tarjeta mastercard y crear una columna que sea la suma de 'PERCEPCION IVA R.G. 2408   1,50 %' y  'PERCEPCION IVA R.G. 2408   3,00 %'

### Guardar como datasets
a- el dataset de la tarjeta visa
b- el dataset de la tarjeta mastercard
c- cuanto dinero se cobra por mes y por sucursal
d- cuanto se cobra por mes
e- crear un lineplot de cuanto se cobra por mes